### PR TITLE
Feature/rafi/create an endpoint to view all questions and answers in specified quiz

### PR DESCRIPTION
**Pull request by rafipriatna**
#### How to reproduce:
HIT GET /courses/:courseId/quizzes/:quizId
Authentication: Access
Role: Instructor / Student who enrolled the course

#### Results:
IF success:
Status Code: 200
```json
{
	"status": "success",
	"data": [
		{
			"id": 1,
			"question": "Apa itu backend?",
			"questionOptions": [
				{
					"id": 1,
					"questionId": 1,
					"option": "Jawaban 1"
				},
				{
					"id": 2,
					"questionId": 1,
					"option": "Jawaban 3"
				},
				{
					"id": 3,
					"questionId": 1,
					"option": "Jawaban 4"
				},
				{
					"id": 4,
					"questionId": 1,
					"option": "Jawaban 5"
				},
				{
					"id": 5,
					"questionId": 1,
					"option": "Jawaban 2"
				}
			]
		},
		....
	],
	"message": "Successfully retrieved all questions and options."
}
```

IF Not enrolled:
Status Code: 400
```json
{
	"status": "fail",
	"message": "Course not enrolled."
}
```

NB: Do unit testing later. Not tested for other conditions

### CODE REVIEW

File: src/controllers/quiz.controller.ts [hunk 1]
```diff
@@ -28,6 +28,23 @@ class QuizController {
...
+
+        const questions = await quizService.ViewAllQuestionsAndOptions(
+            params.courseId,
+            params.quizId,
+            userPayload!);
+
+        return sendResponse(res, {
+            statusCode: StatusCodes.OK,
+            success: true,
+            data: questions,
```
Comment (rahmatsyifana12) [inline src/controllers/quiz.controller.ts]: ini jadiin data: { question } biar jadi list of question gitu

File: src/database/entities/Question.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class Question extends BaseEntity {
     @Column({ name: 'quiz_id' })
     quizId!: number;
 
-    @Column({ length: 128 })
+    @Column({ length: 128, select: false })
```
Comment (rahmatsyifana12) [inline src/database/entities/Question.ts:23]: select: false tuh buat apa ya?

File: src/services/courseEnrollment.service.ts [hunk 3]
```diff
@@ -10,9 +10,8 @@ import { Course } from '../database/entities/Course';
 class CourseEnrollmentService {
 
     async enrollNewCourse(
-        userPayload: UserPayload,
-        { courseId }: AddCourseEnrollmentType) {
-        const userId = userPayload.userId;
+        { courseId }: AddCourseEnrollmentType,
+        userId: UserPayload['userId']) {
```
Comment (rahmatsyifana12) [inline src/services/courseEnrollment.service.ts]: ini jadiin number aja type nya

File: src/services/courseEnrollment.service.ts [hunk 4]
```diff
@@ -39,6 +38,26 @@ class CourseEnrollmentService {
         await CourseEnrollment.save(enrollCourse);
     }
 
+    async getCourseEnrollment(
+        courseId: number,
+        userId: UserPayload['userId']) {
```
Comment (rahmatsyifana12) [inline src/services/courseEnrollment.service.ts]: ini jadiin number aja typenya

**Review by rahmatsyifana12 — CHANGES_REQUESTED** [review]

File: src/database/entities/Question.ts (hunk 2 above)
Comment (rafipriatna) [reply src/database/entities/Question.ts:23]: Biar ga keliatan kunci jawabannya pas di-view all

Comment (rahmatsyifana12) [reply src/database/entities/Question.ts:23]: okok

**Review by rahmatsyifana12 — APPROVED** [review]

**Pull request merged**